grpcService/pkg/user/service.go

```diff
@@ -0,0 +1,104 @@
+package user
+
+import (
+	"context"
+
+	"database/sql"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/google/uuid"
+
+	entities "github.com/timoteoBone/microservice-project/grpcService/pkg/entities"
+	errors "github.com/timoteoBone/microservice-project/grpcService/pkg/errors"
+	mapper "github.com/timoteoBone/microservice-project/grpcService/pkg/mapper"
+)
+
+type Repository interface {
+	GetUser(ctx context.Context, userId string) (entities.User, error)
+	CreateUser(ctx context.Context, user entities.User, newId string) (string, error)
+	DeleteUser(ctx context.Context, userId string) error
+}
+
+type service struct {
+	Repo   Repository
+	Logger log.Logger
+}
+
+func NewService(l log.Logger, r Repository) *service {
+	return &service{r, l}
+}
+
+func (s *service) CreateUser(ctx context.Context, userReq entities.CreateUserRequest) (entities.CreateUserResponse, error) {
+	s.Logger.Log(s.Logger, "request", "create user", "received")
+
+	response := entities.CreateUserResponse{}
+	status := entities.Status{}
+
+	user := mapper.CreateUserRequestToUser(userReq)
+	newId := generateId()
+	genId, err := s.Repo.CreateUser(ctx, user, newId)
+
+	if err != nil {
+		if err.Error() == "Error 1062: Duplicate entry '' for key 'USER.email'" {
```

Comment:
this error  don't have to  be in this layer, and error database  as you have defined in errors. should  pas this to a custom error an here validate if you receive this error  
type UserAlreadyExist.